Dark mode v2.0

pbaity opened the pull request:
There were several goals in these changes:

1. Most importantly, switch to relying mainly on overriding Rocket.Chat CSS variables (defined [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/theme/client/imports/general/variables.css)). Whenever overriding a variable was possible, I preferred that over writing a new CSS rule. More Rocket.Chat variables can be overridden as needed.
2. Remove redundant CSS rules (once the variables were added). This will probably be an ongoing task to make sure we're using as few rules as needed to accomplish the work.
3. Make dark mode easier for others to tweak. Rather than hunting down various rules, I wanted a few key variables to be defined which could a user could tweak to easily and consistently change the dark theme to suit their preferences. This has been partially accomplished and I'll keep working on this in the future.

For now I'm opening the PR because these changes should also fix https://github.com/pbaity/rocketchat-dark-mode/issues/13 and fix https://github.com/pbaity/rocketchat-dark-mode/issues/15.

ankar84 commented:
I just now tested your **dark-mode.css version 2.0** and can tell that #17 is fix defiantly.
And in general all looks very nice! 
Only one bug I could find is in **Administration Panel -> File Upload**
![image](https://user-images.githubusercontent.com/4023037/75417077-2cf19e80-5941-11ea-8798-231bc1551f94.png)
But I think I have that bug and at previous **dark-mode.css** version

brittyazel commented on dark-mode.css:
```diff
@@ -139,12 +175,24 @@ body.dark-mode .sidebar-item > a {
-body.dark-mode .emoji-picker {
-	background-color: var(--rc-color-primary-dark);
+body.dark-mode input {
+	color: var(--color-white);
```
this should probably be var(--color-dark) to solve the issue reported above

brittyazel replied:
err actually that breaks things in the front end. Like the channel search box. hrm.

brittyazel commented on dark-mode.css:
```diff
@@ -139,12 +175,24 @@ body.dark-mode .sidebar-item > a {
 
-body.dark-mode .emoji-picker {
-	background-color: var(--rc-color-primary-dark);
+body.dark-mode input {
```
I think you should delete this element. It ends up messing up either the admin side or the front side. Removing these three lines entirely allows them to nicely default the the appropriate colors

ankar84 replied:
I checked that deletion of 
```
body.dark-mode input {
	color: var(--color-white);
}
```
solved my problem in Admin panel

pbaity replied:
Agreed - not sure why this rule existed to begin with, so I'm removing.

brittyazel commented:
Another fix:

`body.dark-mode .rcx-box--text-style-h1,
body.dark-mode .rcx-subtitle {
	color: var(--color-gray-lightest);
}`

should be:

`body.dark-mode .rcx-box--text-style-h1,
body.dark-mode .rcx-subtitle,
body.dark-mode .rcx-box--text-color-default,
body.dark-mode .rcx-box--text-color-info {
	color: var(--color-gray-lightest);
}`

pbaity commented:
@brittyazel I'll see if I can install the poll app and get that fix working soon, unless you beat me to it. But otherwise I've addressed the comments above, so I'll give one last chance for everyone to review before merging tomorrow.
